DOCSP-49849 - Collation (#628)
mongodb/docs-csharp: source/includes/collation.rst

To configure collation for your operation, create an instance of the
`Collation <{+new-api-root+}/MongoDB.Driver/MongoDB.Driver.Collation.html>`__ class.

The following table describes the parameters that the ``Collation`` constructor accepts.
It also lists the corresponding class property that you can use to read each
setting's value.

.. list-table::
   :header-rows: 1
   :widths: 20 60 20

   * - Parameter
     - Description
     - Class Property

   * - ``locale``
     - | Specifies the International Components for Unicode (ICU) locale. For a list of
         supported locales,
         see :manual:`Collation Locales and Default Parameters </reference/collation-locales-defaults/#supported-languages-and-locales>`
         in the {+mdb-server+} Manual.
       |
       | If you want to use simple binary comparison, use the ``Collation.Simple`` static
         property to return a ``Collation`` object with the ``locale`` set to ``"simple"``.
       | **Data Type**: {+string-data-type+} 
     - ``Locale`` 

   * - ``alternate``

>      - | Optional. Specifies whether the driver considers whitespace and punctuation as base

Reviewer: The other table in this doc set uses italics for `*Optional.*`, but the style guide uses `(*Optional)*`
I'll leave it up to you.

>          characters for purposes of comparison.

Reviewer: I would add an `**Accepted Values:**` section

Author: For all of the C# enum data types, I linked to the relevant API page, which documents the possible options. Similar for the string types linking to a page with the accepted options.

       |
       | **Data Type**: `CollationAlternate <{+new-api-root+}/MongoDB.Driver/MongoDB.Driver.CollationAlternate.html>`__
       | **Default**: ``CollationAlternate.NonIgnorable`` (spaces and punctuation are
         considered base characters)
     - ``Alternate``

   * - ``backwards``
     - | Optional. Specifies whether strings containing diacritics sort from the back of the string
         to the front.
       |
       | **Data Type**: {+bool-data-type+} 
       | **Default**: ``false``
     - ``Backwards``

   * - ``caseFirst``

>      - | Optional. Specifies the sort order of case differences during tertiary level comparisons.

Reviewer: `**Accepted Values:**`

Author: see above

       |
       | **Data Type**: `CollationCaseFirst <{+new-api-root+}/MongoDB.Driver/MongoDB.Driver.CollationCaseFirst.html>`__
       | **Default**: ``CollationCaseFirst.Off``
     - ``CaseFirst``

   * - ``caseLevel``

>      - | Optional. Specifies whether to include case comparison at strength level ``Primary`` or
>          ``Secondary``. 
>        |
>        | When this argument is ``true``:
>        |
>        | - If the strength is ``Primary``, the driver compares base
>            characters and case.
>        | - If the strength is ``Secondary``, the driver compares base
>            characters, diacritics, other secondary differences, and case.
>        |
>        | When this argument is ``false``, the driver doesn't include case comparison at
>          strength level ``Primary`` or ``Secondary``.

Reviewer: If there's a way to add a link to the strength parameter, I would do that here. I'm assuming this is why you were asking about nesting tables the other day, because that would make sense here.
```suggestion
     - | Optional. Specifies whether to include case comparison. 
       |
       | When this argument is ``true``, the behaviour depends on the value of the ``Strength`` property:
       |
       | - If the ``Strength`` property is ``Primary``, the driver compares base
           characters and case.
       | - If the ``Strength`` property is ``Secondary``, the driver compares base
           characters, diacritics, other secondary differences, and case.
       |
       | When this argument is ``false``, the driver doesn't include case comparison.
```

Author: a note about the properties: they're read-only and set directly by these parameters, so i talk about the values of the parameters instead of the values of the properties.

       |
       | **Data Type**: {+bool-data-type+} 
       | **Default**: ``false``
     - ``CaseLevel``

   * - ``maxVariable``

>      - | Optional. Specifies which characters the driver considers ignorable when
>          ``Alternate`` is ``CollationAlternate.Shifted``.

Reviewer: add `**Accepted Values:**`
```suggestion
     - | Optional. Specifies which characters the driver considers ignorable when the
         ``Alternate`` property is ``CollationAlternate.Shifted``.
```

       |
       | **Data Type**: `CollationMaxVariable <{+new-api-root+}/MongoDB.Driver/MongoDB.Driver.CollationMaxVariable.html>`__ 
       | **Default**: ``CollationMaxVariable.Punctuation`` (the driver ignores punctuation
         and spaces)
     - ``MaxVariable``
   
   * - ``normalization``

>      - | Optional. Specifies whether the driver normalizes text as needed.

Reviewer: Maybe add an example

Author: how about a link to the ICU docs

       |
       | Most text doesn't require normalization.
       |
       | **Data Type**: {+bool-data-type+}
       | **Default**: ``false``
     - ``Normalization``

   * - ``numericOrdering``
     - | Optional. Specifies whether the driver compares numeric strings as numbers.
       | 

>        | If this argument is ``true``, the driver compares numeric strings as numbers.
>          For example, "10" is greater than "2". If this argument is ``false`` or excluded,
>          the driver compares numeric strings as strings. For example, "10" is less than "2".

Reviewer: They're not being treated as strings so I would leave out the "s, you could make them `10` and `2` if you want to differentiate them:
```suggestion
       | If this argument is ``true``, the driver compares numeric strings as numbers.
         For example, 10 is greater than 2. 
       | 
       | If this argument is ``false`` or excluded,
         the driver compares numeric strings as strings. For example, "10" is less than "2".
```

Author: they aren't being compared as strings, but they still _are_ strings, so i think the quotes are useful. lmk if this change is clearer.

       |
       | For more information, see :manual:`Collation Restrictions </reference/collation/#restrictions>`
         in the {+mdb-server+} manual.
       |
       | **Data Type**: {+bool-data-type+}
       | **Default**: ``false``
     - ``NumericOrdering``

   * - ``strength``
     - | Optional. Specifies the level of comparison to perform, as defined in the
         `ICU documentation <https://unicode-org.github.io/icu/userguide/collation/concepts.html#comparison-levels>`__.
       |
       | **Data Type**: `CollationStrength <{+new-api-root+}/MongoDB.Driver/MongoDB.Driver.CollationStrength.html>`__
       | **Default**: ``CollationStrength.Tertiary``
     - ``Strength``

> For more information about collation, see the :manual:`Collation <reference/collation>`
> page in the {+mdb-server+} manual.

Reviewer: 404
```suggestion
For more information about collation, see the :manual:`Collation </reference/collation>`
page in the {+mdb-server+} manual.
```
